include/cutils/properties.h

```diff
@@ -35,8 +35,13 @@ extern "C" {
 ** WARNING: system/bionic/include/sys/system_properties.h also defines
 **          these, but with different names.  (TODO: fix that)
 */
+#ifdef PROP_NAME_MAX
```

Comment:
PROPERTY_NAME/VALUE_MAX have to be the same when properties.h is included as when the code is compiled; this is very different from PROPERTIES_PATH which only needs to be defined when compiling properties_nonandroid.c. 

It *would* be possible, I suppose, to adjust the makefiles so that OPT_CFLAGS/OPT_CXXFLAGS are replaced with BASE_CFLAGS/BASE_CXXFLAGS and we add -D flags there, but I think that's really ugly.

What I replaced this with:

#if !ADB_NON_ANDROID
#include <sys/system_properties.h>
#else
+#define PROP_KEY_MAX   32
+#define PROP_VALUE_MAX  92
#endif

to keep this clearly in an existing non-android block?